CLR/Libraries/Windows_Devices/windows_devices_native_Windows_Devices_Spi_SpiDevice.cpp

```diff
@@ -0,0 +1,239 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#include "windows_devices.h"
+
+typedef Library_windows_devices_native_Windows_Devices_Spi_SpiConnectionSettings SpiConnectionSettings;
+
+// NOTE: Must match enum in SpiEnums.cs
+enum SpiMode
+{
+        SpiMode_Mode0 = 0,
+        SpiMode_Mode1,
+        SpiMode_Mode2,
+        SpiMode_Mode3,
+};
+
+HRESULT Library_windows_devices_native_Windows_Devices_Spi_SpiDevice::InitNative___VOID( CLR_RT_StackFrame& stack )
+{
+    TINYCLR_HEADER(); hr = S_OK;
+
+    GPIO_PIN chipSelectPin = GPIO_PIN_NONE;

```

Comment:
We're pretty inconsistent about this one, and I kept having to fight VS to do it.  I think we should stick with the more traditional formatting unless you feel very strongly.